app/src/main/java/org/systers/mentorship/viewmodels/TaskViewModel.kt

```diff
@@ -1,23 +1,65 @@
 package org.systers.mentorship.viewmodels
 
+import android.annotation.SuppressLint
+import android.util.Log
+import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.observers.DisposableObserver
+import io.reactivex.schedulers.Schedulers
 import org.systers.mentorship.MentorshipApplication
 import org.systers.mentorship.R
+import org.systers.mentorship.models.Task
+import org.systers.mentorship.remote.datamanager.RelationDataManager
+import org.systers.mentorship.remote.responses.CustomResponse
+import org.systers.mentorship.utils.CommonUtils
+import retrofit2.HttpException
+import java.io.IOException
+import java.util.concurrent.TimeoutException
 
 /**
  * This class represents the [ViewModel] used for Tasks Screen
  */
-class TasksViewModel: ViewModel() {
-    var taskList = mutableListOf<String>()
-    var completedTaskList = mutableListOf<String>()
+class TasksViewModel : ViewModel() {
+
+    var TAG = TasksViewModel::class.java.simpleName
+
+    private val relationDataManager: RelationDataManager = RelationDataManager()
+
+    var taskList = mutableListOf<Task>()
+    var completedTaskList = mutableListOf<Task>()
+
+    val successfulDelete: MutableLiveData<Boolean> = MutableLiveData()
+    lateinit var message: String
 
     /**
      * This function lists all tasks from the mentorship relation
      */
     fun getTasks() {
-        taskList.add(MentorshipApplication.getContext().getString(R.string.model_task_1))
-        taskList.add(MentorshipApplication.getContext().getString(R.string.model_task_2))
-        taskList.add(MentorshipApplication.getContext().getString(R.string.model_task_3))
+        taskList.add(
```

Comment:
these tasks with very specific values are for testing right?

Reply:
Yup, it is.